index.js

```diff
@@ -2,24 +2,54 @@
 
 // Adapted from https://github.com/markdown-it/markdown-it/blob/fbc6b0fed563ba7c00557ab638fd19752f8e759d/docs/architecture.md
 
-function markdownitLinkAttributes (md, config) {
-  config = config || {}
+var optionKeys = ['ignoreRelative', 'pattern']
+var absoluteUrlPattern = /^([a-z]+:)?\/\//
+
+function markdownitLinkAttributes (md, _config) {
+  // For downward-compatibility, pass in flat options
+  var configs = (Array.isArray(_config) ? _config : [_config || {}]).map(function (config) {
+    var result = { attrs: {} }
+
+    Object.keys(config).forEach(function (key) {
+      if (optionKeys.indexOf(key) > -1) {
+        result[key] = config[key]
+      } else {
+        result.attrs[key] = config[key]
+      }
+    })
+    return result
+  })
+
+  Object.freeze(configs)
 
   var defaultRender = md.renderer.rules.link_open || this.defaultRender
-  var attributes = Object.keys(config)
 
   md.renderer.rules.link_open = function (tokens, idx, options, env, self) {
-    attributes.forEach(function (attr) {
-      var value = config[attr]
-      var aIndex = tokens[idx].attrIndex(attr)
-
-      if (aIndex < 0) { // attr doesn't exist, add new attribute
-        tokens[idx].attrPush([attr, value])
-      } else { // attr already exists, overwrite it
-        tokens[idx].attrs[aIndex][1] = value // replace value of existing attr
+    var href = tokens[idx].attrGet('href')
+
+    var config = configs.find(function (conf) {
```

Comment:
`find` is (unfortunately) [not supported in Internet Explorer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find).

Reply:
WHAT Σ( ° △ °|||)